extensions/schema-compare/src/dialogs/schemaCompareDialog.ts

```diff
@@ -874,13 +874,13 @@ export class SchemaCompareDialog {
 				finalName = c.options.connectionName;
 			}
 
-			// use previously selected server or current connection if there is one
-			if (endpointInfo && !isNullOrUndefined(endpointInfo.serverName) && !isNullOrUndefined(endpointInfo.serverDisplayName)
-				&& c.options.server.toLowerCase() === endpointInfo.serverName.toLowerCase()
-				&& finalName.toLowerCase() === endpointInfo.serverDisplayName.toLowerCase()) {
+			// use current connection else use previously selected server if there is one
+			if (c.connectionId === this.connectionId) {
```

Comment:
what does `this.connectionId` refer to in this dialog? Would it help to have separate variables for source and target connectionIds?

Reply:
this.connectionId is the current active connection. This is used only when schema compare isn't started from a server or db context. There is target and source endpoint info, so don't need a separate connection id for both. This is majorly used when a new connection is added, and helps to keep track of the sequence/list of the active connections available to SC.

Reply:
ah ok got it. Can you add a comment about it being the current active connection where `connectionId ` is declared so it's more clear?